bin/varnishstat/varnishstat_curses.c

```diff
@@ -262,7 +263,12 @@ sample_points(void)
 		pt->t_last = pt->t_cur;
 		pt->t_cur = VTIM_mono();
 
-		if (pt->t_last)
+		if (reset) {
+			pt->chg = 0;
+			pt->ma_10.n = 0;
+			pt->ma_100.n = 0;
+			pt->ma_1000.n = 0;
+		} else if (pt->t_last)
 			pt->chg = ((int64_t)pt->cur - (int64_t)pt->last) /
 			    (pt->t_cur - pt->t_last);
```

Comment:
To be consistent with the block of code from my previous comment, shouldn't it look like this?

```c
if (reset) { // or better: if (reset_averages)
	pt->chg = 0;
	pt->ma_10.n = 0;
	pt->ma_100.n = 0;
	pt->ma_1000.n = 0;
}

if (pt->t_last) {
	pt->chg = ((int64_t)pt->cur - (int64_t)pt->last) /
	    (pt->t_cur - pt->t_last);
}
```

For the hit ratio you reset the 10/100/1000 averages and let them be updated in the same iteration. So for consistency you should give the counters a chance to be updated too.